pkg/mcp/mcp_watch_test.go

```diff
@@ -40,13 +56,22 @@ func (s *WatchKubeConfigSuite) TestNotifiesToolsChange() {
 	s.InitMcpClient()
 	// When
 	s.WriteKubeconfig()
-	notification := s.WaitForNotification(5 * time.Second)
+	notification := s.WaitForNotification(5*time.Second, "notifications/tools/list_changed")
 	// Then
 	s.NotNil(notification, "WatchKubeConfig did not notify")
-	s.True(
-		notification.Method == "notifications/tools/list_changed" || notification.Method == "notifications/prompts/list_changed",
-		"WatchKubeConfig did not notify tools or prompts change, got: %s", notification.Method,
-	)
+	s.Equal("notifications/tools/list_changed", notification.Method, "WatchKubeConfig did not notify tools change")
+}
+
+func (s *WatchKubeConfigSuite) TestNotifiesPromptsChange() {
+	s.T().Skip("Enable when prompt unmarshalling is fixed") // TODO: Fix prompt unmarshalling issue
```

Comment:
if there is an issue with the prompt unmarshalling, can we fix it as part of the same PR here?

Reply:
I will cover that later.
The idea is to simplify the Config unmarshalling, but I haven't analyzed what that would require.
The main purpose of these changes is to unlock #434 which has some changes in this test and shouldn't.